### DISCUSSION TITLE
How to use Web3Provider with wagmi?

### DISCUSSION THREAD

**Discussion by edgardo-mkr** [wevm/wagmi#1105] [Q&A]
Is there a way to create an ethers.provider.web3provider with wagmi? for what i see the provider offered by the useProvider hook is a kind of ethers provider but not an externalProvider which is the type need it for the web3Provider constructor. Our entire application is based in a web3provider and i really would not want to change all that logic

**Comment by tmm** [comment]
Check out [`useSigner`](https://wagmi.sh/docs/hooks/useSigner).

**Reply by skgbafa** [reply]
> Check out [`useSigner`](https://wagmi.sh/docs/hooks/useSigner).

For anyone else landing on this, useSigner returns a nested web3 provider

```javascript
const { data: signer  } = useSigner();
const web3Provider = signer.provider; // of type ethers.provider.web3provider
```

**Reply by appleseed-iii** [reply]
it looks like this returns `ethers.providers.Provider`; I guess depending on usage you can cast it as `ethers.providers.Web3Provider`...
